### PR TITLE
Enable GraphSON 3.0 support by default #1075

### LINKED ISSUE #1075: Support GraphSON 3
> Version 3.0 of GraphSON was first introduced on TinkerPop 3.3.0 and is the default format when not specified as of this version.

From [TinkerPop's I/O docs](http://tinkerpop.apache.org/docs/current/dev/io/#graphson-3d0).

### PR DESCRIPTION
**Pull request by FlorianHockmann**
Turned out that GraphSON 3.0 support was already there with the usual `JanusGraphIoRegistry`. The tests I added in my PR to add support for GraphSON deserialization of predicates (#1061) for some reason just didn't work with GraphSON 3.0 the way I implemented them. It looked like the `RequestMessage` wasn't serialized correctly. So I changed the tests to simply serialize `Bytecode` directly instead and that worked immediately with GraphSON 3.0.
To make sure that GraphSON 3.0 really works completely I tested it manually by connecting with Gremlin.Net and GraphSON 3 configured and I could execute several traversals without any problems, also some that included the new `g:Map` type that wasn't present in GraphSON 2.0.

To ensure that users can utilize GraphSON 3.0 directly when they start the server, I added the serialization config to the respective YAML files. While doing that I also noticed that they still contained some elements in the config that was moved to the `plugins` element in TinkerPop 3.3.0 additionally in its old place. So I also removed those elements.

Fixes #1075

Signed-off-by: Florian Hockmann <fh@florian-hockmann.de>

-----

Thank you for contributing to JanusGraph!

In order to streamline the review of the contribution we ask you
to ensure the following steps have been taken:

### For all changes:
- [x] Is there an issue associated with this PR? Is it referenced in the commit message?
- [x] Does your PR body contain #xyz where xyz is the issue number you are trying to resolve?
- [x] Has your PR been rebased against the latest commit within the target branch (typically `master`)?
- [x] Is your initial contribution a single, squashed commit?

### For code changes:
- [x] Have you written and/or updated unit tests to verify your changes?
- [x] If adding new dependencies to the code, are these dependencies licensed in a way that is compatible for inclusion under [ASF 2.0](http://www.apache.org/legal/resolved.html#category-a)?
- [x] If applicable, have you updated the LICENSE.txt file, including the main LICENSE.txt file in the root of this repository?
- [x] If applicable, have you updated the NOTICE.txt file, including the main NOTICE.txt file found in the root of this repository?

### For documentation related changes:
- [x] Have you ensured that format looks appropriate for the output in which it is rendered?
- [x] If this PR is a documentation-only change, have you added a `[skip ci]`
  tag to the first line of your commit message to avoid spending CPU cycles in
  Travis CI when no code, tests, or build configuration are modified?

### Note:
Please ensure that once the PR is submitted, you check Travis CI for build issues and submit an update to your PR as soon as possible.

### CODE REVIEW

File: janusgraph-dist/src/assembly/static/conf/gremlin-server/gremlin-server-berkeleyje.yaml [hunk 1]
```diff
@@ -10,17 +10,15 @@ scriptEngines: {
...
-    scripts: [scripts/empty-sample.groovy]}}
+               org.apache.tinkerpop.gremlin.jsr223.ScriptFileGremlinPlugin: {files: [scripts/empty-sample.groovy]}}}}
 serializers:
   - { className: org.apache.tinkerpop.gremlin.driver.ser.GryoMessageSerializerV1d0, config: { ioRegistries: [org.janusgraph.graphdb.tinkerpop.JanusGraphIoRegistry] }}
   - { className: org.apache.tinkerpop.gremlin.driver.ser.GryoLiteMessageSerializerV1d0, config: {ioRegistries: [org.janusgraph.graphdb.tinkerpop.JanusGraphIoRegistry] }}
   - { className: org.apache.tinkerpop.gremlin.driver.ser.GryoMessageSerializerV1d0, config: { serializeResultToString: true }}
   - { className: org.apache.tinkerpop.gremlin.driver.ser.GraphSONMessageSerializerGremlinV1d0, config: { ioRegistries: [org.janusgraph.graphdb.tinkerpop.JanusGraphIoRegistryV1d0] }}
   - { className: org.apache.tinkerpop.gremlin.driver.ser.GraphSONMessageSerializerGremlinV2d0, config: { ioRegistries: [org.janusgraph.graphdb.tinkerpop.JanusGraphIoRegistry] }}
   - { className: org.apache.tinkerpop.gremlin.driver.ser.GraphSONMessageSerializerV1d0, config: { ioRegistries: [org.janusgraph.graphdb.tinkerpop.JanusGraphIoRegistryV1d0] }}
```
[the remark this answers is not in the document]
Comment (FlorianHockmann) [reply janusgraph-dist/src/assembly/static/conf/gremlin-server/gremlin-server-berkeleyje.yaml]: Which of these would you remove? TinkerPop only has serializers for the latest versions configured but that forces clients to also use the latest version. That would make an upgrade to JanusGraph 0.3.0 a bit problematic in my opinion as all clients would have to update at the same time as they couldn't use GraphSON v3 before and after the update they can't use anything < v3.

We could maybe instead list the newest serializers at the top and the older ones after a comment that explains that they are kept for backwards compatibility.

[the remark this answers is not in the document]
Comment (pluradj) [reply janusgraph-dist/src/assembly/static/conf/gremlin-server/gremlin-server-berkeleyje.yaml]: +1. I like the idea of putting the newest at the top and adding comments.

[the remark this answers is not in the document]
Comment (pluradj) [reply janusgraph-dist/src/assembly/static/conf/gremlin-server/gremlin-server-berkeleyje.yaml]: perhaps add `GryoMessageSerializerV3d0` also

[the remark this answers is not in the document]
Comment (FlorianHockmann) [reply janusgraph-dist/src/assembly/static/conf/gremlin-server/gremlin-server-berkeleyje.yaml]: Sure, Gryo should also be there in the most recent version. I just pushed a commit with those changes.

**Comment by pluradj** [comment]
@JanusGraph/committers could we get some review and approval on this PR? There was a recent [`janusgraph-dev`](https://groups.google.com/d/msg/janusgraph-dev/MYoyTufqomo/KWA7tnE6AgAJ) discussion and this PR definitely makes it easier for GLVs to connect to JanusGraph. If we can get this merged, I'll spin up another RC.

**Comment by pluradj** [comment]
thanks @jerryjch

**Pull request merged**